### PR TITLE
Feature/complete actions with objective

### LINKED ISSUE #524: Mark actions as completed when completing an objective
**Is your feature request related to a problem? Please describe.**
When we mark an objective as completed, we don't mark the actions under that objective as completed as well.

**Describe the solution you'd like**
We should mark all of the actions of an objective as completed when we mark the objective as completed

**Describe alternatives you've considered**
As @lucca65 pointed out in #519, we might need a loading modal to display the progress

**Additional context**
Came up in #519

### PR DESCRIPTION
**Pull request by NeoVier**
## What issue does this PR close
Closes #524 

## Changes Proposed ( a list of new changes introduced by this PR)
- Reload community data when user enters `/community/objectives`
- Update the text that says completing an objective is not reversible
- When an objective is marked as complete, do the same for all the actions under that objective (and show a loading screen while doing so)

## How to test ( a list of instructions on how to test this PR)
1. Be logged in on the admin account for a community
2. Go to `/community/objectives` (Dashboard > Cog icon (top right) > Objectives and Actions)
3. If there aren't any objectives with (non-completed) actions, create one, and fill it with actions
4. Find an objective that has a bunch of actions (the more the merrier!), click on it and click on "Edit objective"
5. Click on "Complete objective"
6. See the new loading screen
7. Be redirected back to `/community/objectives`
8. Check if all the actions under the objective (and the objective itself) have a "Completed" badge

### CODE REVIEW

**Comment by NeoVier** [comment]
@lucca65 since we created cambiatus/improvement-proposals#44, I've settled for just showing a `Feedback` when completing an objective. It goes exactly as you suggested (per action):
1. We try to complete the action
2. If it fails, we try again, up to 3 times in total
3. Once every action has been marked as complete (or failed 3 times), we mark the objective as complete and if there are any errors, we log it to sentry:
2021-07-12-213733_661x496_scrot

4. Redirect the user to `/community/objectives`.
 
Is this good enough for us to merge this? I think it's taken way too long already and it works fine as is.

For now the logging to sentry is very basic (it's not really something we already do - it doesn't fit with any of the existing options, so I created a `Generic` one), as we have #552 to improve that

File: src/elm/Log.elm [hunk 1]
```diff
@@ -99,6 +100,10 @@ send toStrs (Log a) =
             )
                 |> logDebug
 
+        Generic err ->
```
[the remark this answers is not in the document]
Comment (NeoVier) [reply src/elm/Log.elm]: Done!

**Review by lucca65 — DISMISSED** [review]

**Comment by juramos-2020** [comment]
I did the test with an objective that had only two actions, I marked it as completed, the confirmation modal appeared, the loading screen also appeared and I was directed to /community/objectives. The "Completed" badge only appears on the objective.

After that I  created an objective to test with several actions, but it was giving an error to create new actions

image

**Comment by NeoVier** [comment]
@juramos-2020 that issue should be fixed now, can you try again?

**Comment by juramos-2020** [comment]
Hi

I managed to create the actions now and everything worked out with the test. 👌

I just found that the loading modal with the progress bar appeared too fast. The progress bar completed in a "blink of an eye".

And another adjustment that can be checked at another time is the option to put the expiration date on the stock. For some reason the "action validity" field is not recognizing the date informed:
image

**Comment by NeoVier** [comment]
> I just found that the loading modal with the progress bar appeared too fast. The progress bar completed in a "blink of an eye".

That's because we're sending all the actions to be completed at once (as individual transactions, but all of the transactions are sent at once), and they all take around the same time to come back... @lucca65 this still solves the blockchain problem, right? Or should we do one, and then do the next, and then the next...?

Also, I noticed the page kind of freezes (you can see the loading animation drop a lot of frames) when we get a response from and eos transaction (I'm 90% sure that's the cause). I guess there's not much we can do about that, just putting it out there :sweat_smile: 

> And another adjustment that can be checked at another time is the option to put the expiration date on the stock. For some reason the "action validity" field is not recognizing the date informed:

@juramos-2020 there's an error message that says "Use the MM/DD/AAAA format" there, meaning you should use 08/22/2021 instead of 22/08/2021. Anyway, we have a calendar component in our app, we should probably use that here as well, I'll open an issue for that

**Comment by juramos-2020** [comment]
Oi @NeoVier 

refiz o teste agora e vi que o progresso da barra caminhou mais lentamente. Ficou bom. 😉

https://user-images.githubusercontent.com/67649886/125634208-eab4103e-78a3-4032-bba0-0c43ea511c92.mp4

**Comment by lucca65** [comment]
> That's because we're sending all the actions to be completed at once (as individual transactions, but all of the transactions are sent at once), and they all take around the same time to come back... @lucca65 this still solves the blockchain problem, right? Or should we do one, and then do the next, and then the next...?

All transactions at once will make us stumble on the same problem, because RAM/CPU have a maximum value. Usually we hit this limit with validations but hardly with action edition. I would say its ok to keep it like that, sending in batches, but the ideal would be to send one by one

**Review by lucca65 — APPROVED** [review]

**Pull request merged**